src/Microsoft.VisualStudio.ProjectSystem.Managed/Text/StringHash.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Diagnostics.Contracts;
+
+namespace Microsoft.VisualStudio.Text
+{
+    internal static class StableStringHash
```

Comment:
Can you add a comment here explaining where you pulled this implementation from? I know it's in the PR description, but having it here in the code would be helpful.

Also explain that it should not be changed, as doing so would break asset compatibility.

Reply:
comment added.

done